### PR TITLE
Add Food Cards on Restaurant Page

### PR DESCRIPTION
**Pull request by Remos96**
Restaurant Page has food cards now with their associated names, images, descriptions, and prices.
image

Styled it in such a way that browser rescaling/resizing (espcially in small screen sizes) fits content nicely. See next two images as examples

image

image

### CODE REVIEW

**Review by JoshuaRamc — COMMENTED** [review]
I was on call with Omer doing this, approved his stylistic choices as it scales with screen size.

**Review by manav809 — APPROVED** [review]
lgtm

File: client/src/pages/Restaurant/FoodList/FoodCard.js [hunk 1]
```diff
@@ -1,5 +1,57 @@
...
+      fontSize: "1rem",
+      fontWeight: "bold",
+      overflow: "hidden",
+      whitespace: "nowrap",
+      textOverflow: "ellipsis",
+      height: "30%",
+    },
+    foodPrice: {
+      fontSize: "1rem",
+      marginLeft: "auto",
```
Comment (DLi7077) [inline client/src/pages/Restaurant/FoodList/FoodCard.js:26]: Nice

File: client/src/pages/Restaurant/FoodList/FoodCard.js [hunk 2]
```diff
@@ -1,5 +1,57 @@
-function FoodCard() {
-  return <div>FoodCard</div>;
+function FoodCard({ name, price, description, imageUrl }) {
+  const classes = {
+    cardContainer: {
+      boxSizing: "border-box",
+      width: "100%",
+      height: "68px",
+      border: "2px solid #5a5a5a",
+      padding: "2px",
```
Comment (DLi7077) [inline client/src/pages/Restaurant/FoodList/FoodCard.js:8]: Use rem

File: client/src/pages/Restaurant/FoodList/FoodCard.js [hunk 3]
```diff
@@ -1,5 +1,57 @@
...
+      display: "flex",
+      gap: "0.25rem",
+      backgroundColor: "hsl(35,0%,97%)",
+    },
+    foodInfo: {
+      maxWidth: "740px",
```
Comment (DLi7077) [inline client/src/pages/Restaurant/FoodList/FoodCard.js:14]: Do we need this?

File: client/src/pages/Restaurant/FoodList/FoodCard.js [hunk 4]
```diff
@@ -1,5 +1,57 @@
...
+    },
+    foodName: {
+      fontSize: "1rem",
+      fontWeight: "bold",
+      overflow: "hidden",
+      whitespace: "nowrap",
+      textOverflow: "ellipsis",
+      height: "30%",
```
Comment (DLi7077) [inline client/src/pages/Restaurant/FoodList/FoodCard.js:22]: fontSize 1rem not needed

Remove bold
What's the height for?

File: client/src/pages/Restaurant/FoodList/FoodCard.js [hunk 5]
```diff
@@ -1,5 +1,57 @@
...
+      fontSize: "1rem",
+      marginLeft: "auto",
+      padding: "15px",
+    },
+    foodDescription: {
+      fontSize: "0.8rem",
+      overflow: "auto",
+      whiteSpace: "normal",
+      textOverflow: "ellipsis",
+      height: "45px",
```
Comment (DLi7077) [inline client/src/pages/Restaurant/FoodList/FoodCard.js:34]: Make this like .75 rem with color gray

What's the height for?

File: client/src/pages/Restaurant/FoodList/FoodCard.js [hunk 6]
```diff
@@ -1,5 +1,57 @@
-function FoodCard() {
-  return <div>FoodCard</div>;
+function FoodCard({ name, price, description, imageUrl }) {
+  const classes = {
+    cardContainer: {
+      boxSizing: "border-box",
+      width: "100%",
+      height: "68px",
```
Comment (DLi7077) [inline client/src/pages/Restaurant/FoodList/FoodCard.js:6]: Please +1 the height

File: client/src/pages/Restaurant/FoodList/FoodCard.js [hunk 7]
```diff
@@ -1,5 +1,57 @@
...
+      fontSize: "0.8rem",
+      overflow: "auto",
+      whiteSpace: "normal",
+      textOverflow: "ellipsis",
+      height: "45px",
+    },
+    image: {
+      backgroundColor: "#F0EAD6",
+      width: "42px",
+      objectFit: "contain",
```
Comment (DLi7077) [inline client/src/pages/Restaurant/FoodList/FoodCard.js:39]: Why do we need a background color?
Looks like it's missing height 100%

File: client/src/pages/Restaurant/FoodList/FoodCard.js [hunk 8]
```diff
@@ -1,5 +1,57 @@
...
+      borderRadius: "5px",
+      margin: "2px",
```
Comment (DLi7077) [inline client/src/pages/Restaurant/FoodList/FoodCard.js:41]: Use rem

Comment (DLi7077) [reply client/src/pages/Restaurant/FoodList/FoodCard.js:41]: Also why do we need margin around the image? Flex gap already handles it

File: client/src/pages/Restaurant/FoodList/index.js [hunk 9]
```diff
@@ -1,5 +1,27 @@
...
+    justifyContent: "center",
+    gap: "1rem",
+  },
+};
+
+function FoodList({ foods }) {
+  return (
+    <div style={classes.foodList}>
+      {foods.map((food) => (
+        <FoodCard
```
Comment (DLi7077) [inline client/src/pages/Restaurant/FoodList/index.js:16]: Missing key here. If you inspect, you should expect React complaining about it. Just use a temp key for now. In the future, we'll use the food Id from the database (need to fix the food presenter from the server)

**Review by DLi7077 — CHANGES_REQUESTED** [review]
Try regular text name and smaller gray text description.

Make the image height 100% with the fixed width

File: client/src/pages/Restaurant/FoodList/FoodCard.js (hunk 4 above)
Comment (Remos96) [reply client/src/pages/Restaurant/FoodList/FoodCard.js:22]: _Height_: It allows the name/title of the food to get cut off/dissapear when resizing to a smaller window. Doesn't work without it

File: client/src/pages/Restaurant/FoodList/FoodCard.js (hunk 5 above)
Comment (Remos96) [reply client/src/pages/Restaurant/FoodList/FoodCard.js:34]: Similar reasoning to the foodName height but the height here is for making sure the food description doesn't spill over outside the food card. Instead, it now provides a little scrollbar for long descriptions.

File: client/src/pages/Restaurant/FoodList/FoodCard.js (hunk 7 above)
Comment (Remos96) [reply client/src/pages/Restaurant/FoodList/FoodCard.js:39]: When an image doesn't fit its container (not all images are sqaure), I put a light easy on the eyes background color to sort of "fill-in" the rest of the div and it provides some consistant sizing of the image area I feel.

File: client/src/pages/Restaurant/FoodList/FoodCard.js (hunk 8 above)
Comment (Remos96) [reply client/src/pages/Restaurant/FoodList/FoodCard.js:41]: Without it, images move a bit towards the top left corner of the card

File: client/src/pages/Restaurant/FoodList/FoodCard.js [hunk 10]
```diff
@@ -1,5 +1,56 @@
...
+      border: "2px solid #5a5a5a",
+      padding: "0.125rem",
+      display: "flex",
+      gap: "0.25rem",
+      backgroundColor: "hsl(35,0%,97%)",
+    },
+    foodName: {
+      overflow: "hidden",
+      whitespace: "nowrap",
+      textOverflow: "ellipsis",
```
Comment (DLi7077) [inline client/src/pages/Restaurant/FoodList/FoodCard.js:15]: Looks like this cuts off the food name if it gets too long.. Is this intended? If it doesn't cut it off, then prob not needed

File: client/src/pages/Restaurant/FoodList/FoodCard.js [hunk 11]
```diff
@@ -1,5 +1,56 @@
...
+    },
+    foodPrice: {
+      fontSize: "1rem",
+      marginLeft: "auto",
+      padding: "15px",
```
Comment (DLi7077) [inline client/src/pages/Restaurant/FoodList/FoodCard.js:20]: nit: use padding 1rem

File: client/src/pages/Restaurant/FoodList/FoodCard.js [hunk 12]
```diff
@@ -1,5 +1,56 @@
...
+      display: "flex",
+      flexDirection: "column",
+      justifyContent: "center"
+    },
+    foodDescription: {
+      fontSize: "0.75rem",
+      color: "#3A3A3A",
+      overflow: "auto",
+      whiteSpace: "normal",
+      textOverflow: "ellipsis",
```
Comment (DLi7077) [inline client/src/pages/Restaurant/FoodList/FoodCard.js:30]: textOverflow wont happen if you're allowing overflow

File: client/src/pages/Restaurant/FoodList/FoodCard.js [hunk 13]
```diff
@@ -1,5 +1,56 @@
...
+    },
+    image: {
+      minWidth: "52px",
+      width: "52px",
+      height: "52px",
+      objectFit: "cover",
+      borderRadius: "2px",
+    },
+  };
+  return (
```
Comment (DLi7077) [inline client/src/pages/Restaurant/FoodList/FoodCard.js:44]: nit: new line before return

File: client/src/pages/Restaurant/FoodList/index.js [hunk 14]
```diff
@@ -1,5 +1,28 @@
-function FoodList() {
-  return <div>FoodList</div>;
+import FoodCard from "./FoodCard";
+
+const classes = {
+  foodList: {
+    display: "flex",
+    flexWrap: "wrap",
+    justifyContent: "center",
```
Comment (DLi7077) [inline client/src/pages/Restaurant/FoodList/index.js:7]: why do we need to center this? isn't this list going downwards?

**Review by DLi7077 — CHANGES_REQUESTED** [review]

**Review by DLi7077 — APPROVED** [review]
good work

**Pull request merged**